### ISSUE TITLE
能支持 wsl 吗？

### ISSUE THREAD

**Issue by hulucc** [ji4ozhu/kkcap#3]

**Comment by ji4ozhu** [comment]
wsl？？？是什么，

**Comment by hulucc** [comment]
https://docs.microsoft.com/en-us/windows/wsl/about

**Comment by ji4ozhu** [comment]
> https://docs.microsoft.com/en-us/windows/wsl/about
> proxifier 工作在LSP层，支持不了
你试试进程那输入 wsl 启动的进程名。应该是可以的

**Comment by hulucc** [comment]
启动进程运行完就结束了，网络不是通过那个进程走的

**Comment by ji4ozhu** [comment]
> 启动进程运行完就结束了，网络不是通过那个进程走的

kkcap 的界面ui上有没有显示那个进程和流量。
比如你可以试试运行wsl里的wget 之类的做测试。

**Comment by xmdmk** [comment]
> 启动进程运行完就结束了，网络不是通过那个进程走的

去任务管理器里看那个进程的详细信息，那里有运行时的名称，看下和你填的是不是一样的。

**Comment by hulucc** [comment]
> > 启动进程运行完就结束了，网络不是通过那个进程走的
> 
> 去任务管理器里看那个进程的详细信息，那里有运行时的名称，看下和你填的是不是一样的。

pico process，和普通process不一样的

**Comment by ji4ozhu** [comment]
你用其他方式找出来哪个进程联网，代理那个进程就可以了

**Comment by hulucc** [comment]
> 你用其他方式找出来哪个进程联网，代理那个进程就可以了

https://slab.qq.com/news/tech/1460.html
pico 进程，代理不了

**Comment by ji4ozhu** [comment]
> > 你用其他方式找出来哪个进程联网，代理那个进程就可以了
> 
> https://slab.qq.com/news/tech/1460.html
> pico 进程，代理不了
![image](https://raw.githubusercontent.com/ji4ozhu/imagefiles/master/wget.PNG)

兄die.是可以代理 wsl 进程的哦。只要你的规则里输入的 联网进程名是正确的就可以了

**Comment by hulucc** [comment]
666

**Comment by hulucc** [comment]
小弟试了下确实能捕捉到tcp的请求，但是dns似乎都没走，所以在wsl里curl www.google.com失败
![image](https://user-images.githubusercontent.com/4343900/53013010-92ce5f00-347f-11e9-910b-6641795d5f65.png)

**Comment by ji4ozhu** [comment]
> 小弟试了下确实能捕捉到tcp的请求，但是dns似乎都没走，所以在wsl里curl [www.google.com失败](http://www.google.com%E5%A4%B1%E8%B4%A5)
> ![image](https://user-images.githubusercontent.com/4343900/53013010-92ce5f00-347f-11e9-910b-6641795d5f65.png)

建议使用直连ss不要用127.0.0.1 的本地转发。

**Comment by hulucc** [comment]
辛苦啦~
用socks5是因为最后会用v2ray(WebSocket + TLS + CDN比较安全，ss不支持

**Comment by hulucc** [comment]
试了下 20190227a，dns还是解析不上（看changelog里有dns forward

![image](https://user-images.githubusercontent.com/4343900/54902666-ff23fe80-4f14-11e9-89c5-a9ce90650065.png)

![image](https://user-images.githubusercontent.com/4343900/54902679-03501c00-4f15-11e9-8a86-decb3a1f857c.png)

**Comment by florianmonfort** [comment]
@hulucc hi, I don't really understand how to use this with WSL? Is it possible?

**Comment by hulucc** [comment]
> @hulucc hi, I don't really understand how to use this with WSL? Is it possible?

Yes, it works, but not very well. And it's a closed source software, so nothing we can do here.

**Comment by a5066987** [comment]
WSL直接用proxychains4。这是我折腾很久之后的最终解决方案
